src/jit/codegenxarch.cpp

```diff
@@ -2158,12 +2158,195 @@ void CodeGen::genMultiRegCallStoreToLocal(GenTree* treeNode)
         offset += genTypeSize(type);
     }
 
-    varDsc->lvRegNum            = REG_STK;
+    varDsc->lvRegNum = REG_STK;
 #else  // !UNIX_AMD64_ABI && !_TARGET_X86_
     assert(!"Unreached");
 #endif // !UNIX_AMD64_ABI && !_TARGET_X86_
 }
 
+//------------------------------------------------------------------------
+// genAllocLclFrame: Probe the stack and allocate the local stack frame: subtract from SP.
+//
+void CodeGen::genAllocLclFrame(unsigned frameSize, regNumber initReg, bool* pInitRegZeroed, regMaskTP maskArgRegsLiveIn)
+{
+    assert(compiler->compGeneratingProlog);
+
+    if (frameSize == 0)
+    {
+        return;
+    }
+
+    const target_size_t pageSize = compiler->eeGetPageSize();
+
+    if (frameSize == REGSIZE_BYTES)
+    {
+        // Frame size is the same as register size.
+        inst_RV(INS_push, REG_EAX, TYP_I_IMPL);
+    }
+    else if (frameSize < pageSize)
+    {
+        // Frame size is (0x0008..0x1000)
+        inst_RV_IV(INS_sub, REG_SPBASE, frameSize, EA_PTRSIZE);
+    }
+    else if (frameSize < compiler->getVeryLargeFrameSize())
+    {
+        // Frame size is (0x1000..0x3000)
```

Comment:
Do we actually need `getVeryLargeFrameSize`? It's kind of strange to call a function for this and then have a comment and even actual code that assumes that the function returns a certain value.

Reply:
Interesting that the function returns `2 * pageSize` on ARM and `3 * pageSize` on other platforms. So not only these comments are wrong but there is also a deadcode in ARM genAllocLclFrame in `else if (frameSize < compiler->getVeryLargeFrameSize())` case.

For now, I would prefer to keep these changes as they are and open a separate PR to address all these issues with `getVeryLargeFrameSize`

Reply:
> So not only these comments are wrong but there is also a deadcode in ARM genAllocLclFrame in else if (frameSize < compiler->getVeryLargeFrameSize()) case.

Right, didn't even notice that. That's one of the side effects of attempting to share code between architectures at any cost :)